GliaWidgetsTests/Sources/CallVisualizer/Coordinator/CallVisualizerCoordinatorTests.swift

```diff
@@ -0,0 +1,141 @@
+import Foundation
+import XCTest
+@testable import GliaWidgets
+
+final class CallVisualizerCoordinatorTests: XCTestCase {
+    var coordinator: CallVisualizer.Coordinator!
+    var viewController = UIViewController()
+
+    override func setUp() {
+        coordinator = .init(environment: .mock)
+    }
+
+    func test_showVisitorCodeViewController() throws {
+        let scene = try XCTUnwrap(UIApplication.shared.connectedScenes.first as? UIWindowScene)
+        let window = scene.windows.first
+        let oldRootViewController = window?.rootViewController
+        window?.rootViewController = viewController
+        defer { window?.rootViewController = oldRootViewController }
+
+        coordinator.showVisitorCodeViewController(by: .alert(viewController))
+        
+        XCTAssertTrue(viewController.presentedViewController is CallVisualizer.VisitorCodeViewController)
+    }
+
+    func test_handleAcceptedUpgrade() {
+        var calledEvents: [CallVisualizer.Coordinator.DelegateEvent] = []
+        coordinator.environment.eventHandler = { calledEvents.append($0) }
+        coordinator.handleAcceptedUpgrade()
+
+        XCTAssertTrue(calledEvents.contains(.maximized))
+    }
+
+    func test_handleEngagementRequestAccepted() throws {
+        let site = CoreSdkClient.Site(
+            id: .mock, defaultOperatorPicture: nil,
+            alwaysUseDefaultOperatorPicture: false,
+            allowedFileSenders: try .mock(),
+            maskingRegularExpressions: [],
+            visitorAppDefaultLocale: "",
+            mobileConfirmDialogEnabled: false,
+            mobileObservationIndicationEnabled: true,
+            mobileObservationVideoFps: try videoFps(),
+            mobileObservationEnabled: true
+        )
+
+        coordinator.environment.fetchSiteConfigurations = { callback in
+            callback(.success(site))
+        }
+
+        let answer = Command<Bool> { boolean in
+            XCTAssertTrue(boolean)
```

Comment:
This assert will only be called if `answer` command is called. To avoid missing this command call in unit tests I think it would be better to introduce empty array of bools, and append `boolean ` to it and assert it out of the closure like this:

```
var answers: [Bool] = []
let answer = Command<Bool> { boolean in
  answers.append(boolean)
}

XCTAssertEqual(answers, [true])
```

Reply:
Fixed 21298a0a635189b8bb2191b3dc1b34e4e621e7be